src/transformers/models/albert/modeling_albert.py

```diff
@@ -249,10 +249,14 @@ def forward(
             else:
                 token_type_ids = torch.zeros(input_shape, dtype=torch.long, device=self.position_ids.device)
 
+            local_rank = os.getenv("LOCAL_RANK")
+            if local_rank is not None:
+                token_type_ids = token_type_ids.to(local_rank)
+
```

Comment:
@sgugger, this is new to `transformers` - the models now become rank aware. should we handle this in a clean way by extending the model's API to get local_rank?

Reply:
Ok Firstly, We can start models that doesn't have token type ids like gpt2 or gpt neo. 
as you said, It's easier.

Reply:
What I suggested is that we work out all the kinks in **all** models that are unique, and any similar models worked out afterwards. So if you're finding yourself replicating the same code - it's not unique - if that's a good rough guideline to follow.

Reply:
Sorry. I think I misunderstood what you said. (As you know, I'm not good at English because I've never lived in an English-speaking country.)

Reply:
I think it is better to define a new abstract class or write a utility function than to use the os module directly.

Reply:
Please don't worry about that, even those with great grasp of a certain language misunderstand each other very often. It's normal. 

So we will just try to explain again in a different way until we understand each other. And code samples are the best way to communicate at times ;)

So to try again, what I was proposing is to pick all the models that require unique handling. And once those are figured out we can replicate all the similar ones. e.g. we have some models which are 95% identical to 5-10 others.

Alternatively, we can also take a very different approach. We can pick just one model - say T5 or GPT2 and completely port it, and then do a few more models, etc, etc. The drawback in this approach is that it'd be more difficult to see how to generalize, but I think getting one model working sooner is more practical and we can get users to start experimenting and report flaws sooner than later.

@sgugger, what's your take on this? Do the integration in stages across all models? Or do 1-2 models, and then repeat?